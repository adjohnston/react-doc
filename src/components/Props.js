import React, { PropTypes } from 'react'
import Subtitle from 'components/styled/Subtitle'

//    requiredProp : [string|object] -> element
const requiredProp = prop => (
  typeof prop === 'object' ? (
    <b>{prop[0]}</b>
  ) : prop
)

const Props = ({props}) => props && (
  <section>
    <Subtitle>
      Props
    </Subtitle>

    <ul>
      {Object.keys(props).map(prop => (
        <li
          key={prop}>
          {prop}: {requiredProp(props[prop])}
        </li>
      ))}
    </ul>
  </section>
) || null

Props.propTypes = {
  props: PropTypes.object
}

export default Props
