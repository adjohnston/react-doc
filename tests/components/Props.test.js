import React from 'react'
import renderer from 'react-test-renderer'
import Props from 'components/Props'

const mockState = {
  id: 'just-a-string-id',
  onClick: () => {},
  value: 20,
  description: 'Something something'
}

const mockProps = {
  id: {
    type: 'string'
  },

  onClick: {
    type: 'func',
    required: true
  },

  value: {
    type: 'number',
    defaultValue: 10
  },

  description: {
    type: 'string',
    description: 'Just a basic description'
  }
}

test('renders null without props', () => {
  const tree = renderer
    .create(
      <Props
        state={mockState}
        setProp={() => {}} />
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})

test('renders null without state', () => {
  const tree = renderer
    .create(
      <Props
        props={mockProps}
        setProp={() => {}} />
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})

test('renders null without state and props', () => {
  const tree = renderer
    .create(
      <Props
        setProp={() => {}} />
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})

test('renders correctly', () => {
  const tree = renderer
    .create(
      <Props
        setProp={() => {}}
        state={mockState}
        props={mockProps} />
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
