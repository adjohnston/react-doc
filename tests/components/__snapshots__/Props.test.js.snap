// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section>
  <h2>
    Props
  </h2>
  <ul>
    <li>
      <div>
        id
      </div>
      <div>
        <b>
          string
        </b>
      </div>
      <input
        className="stncl__input"
        onChange={[Function]}
        type="text"
        value="just-a-string-id"
      />
    </li>
    <li>
      <div>
        onClick
      </div>
      <div>
        <b>
          func
        </b>
        <i>
           - is required
        </i>
      </div>
    </li>
    <li>
      <div>
        value
      </div>
      <div>
        <b>
          number
        </b>
      </div>
      <input
        className="stncl__input"
        onChange={[Function]}
        type="number"
        value={20}
      />
    </li>
    <li>
      <div>
        description
      </div>
      <div>
        <b>
          string
        </b>
      </div>
      <input
        className="stncl__input"
        onChange={[Function]}
        type="text"
        value="Something something"
      />
    </li>
  </ul>
</section>
`;

exports[`renders null without props 1`] = `null`;

exports[`renders null without state 1`] = `null`;

exports[`renders null without state and props 1`] = `null`;
